tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/GitHubActions/GitHubClientFactory.cs

```diff
@@ -0,0 +1,181 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Sdk.Tools.PipelineWitness.Configuration;
+using Azure.Sdk.Tools.PipelineWitness.Utilities;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.JsonWebTokens;
+using Microsoft.IdentityModel.Tokens;
+using Octokit;
+
+namespace Azure.Sdk.Tools.PipelineWitness.GitHubActions;
+
+public class GitHubClientFactory
+{
+    private readonly TimeSpan processTimeout = TimeSpan.FromSeconds(13);
```

Comment:
Why 13 seconds?

Reply:
That's a great question.   Much of the CLI interaction was lifted from [AzureCliCredential](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/src/Credentials/AzureCliCredential.cs#L80) and their default is 13 seconds